cloudinary_cli/utils/api_utils.py

```diff
@@ -96,12 +98,34 @@ def handle_api_command(
 
     args, kwargs = parse_args_kwargs(func, params[1:]) if len(params) > 1 else ([], {})
 
+    if recursive and cursor is None:
+        raise Exception("Using the `--recursive` flag requires a `--cursor` flag.")
+
     res = func(*args, **{
         **kwargs,
         **{k: v for k, v in optional_parameter},
         **{k: parse_option_value(v) for k, v in optional_parameter_parsed},
     })
 
+    if recursive and \
+            confirm_action("Using the --recursive flag will use multiple API calls.\n" +
+                           f"You currently have {res.rate_limit_remaining} Admin API calls remaining. Continue?"):
+        all_results = res
+
+        while True:
+            res = func(*args, **{
+                **kwargs,
```

Comment:
It would be better to remove this code duplication by merging everything to `kwargs` dict, or create another dict called `all_kwargs` and put everything in it. No need to perform this operation on every loop. It would also improve readability of the code.